101-data-factory-v2-blob-to-blob-copy/azuredeploy.json

```diff
@@ -1,209 +1,193 @@
+	"$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+	"contentVersion": "1.0.0.0",
+	"parameters": {
+		"dataFactoryName": {
+			"type": "string",
+			"defaultValue": "[concat('datafactory', uniqueString(resourceGroup().id))]",
+			"metadata": {
+				"description": "Data Factory Name"
+			}
+		},
+		"location": {
+			"type": "string",
+			"defaultValue": "[resourceGroup().location]",
+			"metadata": {
+				"description": "Location of the data factory. Currently, only East US, East US 2, and West Europe are supported."
+			}
+		},
+		"storageAccountName": {
+			"type": "string",
+			"defaultValue": "[concat('storage', uniqueString(resourceGroup().id))]",
+			"metadata": {
+				"description": "Name of the Azure storage account that contains the input/output data."
+			}
+		},
+		"blobContainer": {
+			"type": "string",
+			"defaultValue": "[concat('blob', uniqueString(resourceGroup().id))]",
+			"metadata": {
+				"description": "Name of the blob container in the Azure Storage account."
+			}
+		}
+	},
+	"variables": {
+		"factoryId": "[resourceId('Microsoft.DataFactory/factories/', parameters('dataFactoryName'))]",
+		"storageAccountId": "[resourceId('Microsoft.Storage/storageAccounts', parameters('storageAccountName'))]",
+		"storageLinkedService": "[concat(variables('factoryId'), '/linkedServices/ArmtemplateStorageLinkedService')]",
+		"datasetIn": "[concat(variables('factoryId'), '/datasets/ArmtemplateTestDatasetIn')]",
+		"datasetOut": "[concat(variables('factoryId'), '/datasets/ArmtemplateTestDatasetOut')]"
+	},
+	"resources": [{
+			"type": "Microsoft.Storage/storageAccounts",
+			"apiVersion": "2019-06-01",
+			"name": "[parameters('storageAccountName')]",
+			"location": "[parameters('location')]",
+			"sku": {
+				"name": "Standard_LRS"
+			},
+			"kind": "StorageV2",
+			"properties": {},
+			"resources": [
+				{
+					"type": "blobServices/containers",
+					"apiVersion": "2019-06-01",
+					"name": "[concat('default/', parameters('blobContainer'))]",
+					"dependsOn": [
+						"[parameters('storageAccountName')]"
+					]
+				}
+			]
+		},
+		{
+			"type": "Microsoft.DataFactory/factories",
+			"name": "[parameters('dataFactoryName')]",
+			"apiVersion": "2018-06-01",
+			"location": "[parameters('location')]",
+			"properties": {},
+			"identity": {
+				"type": "SystemAssigned"
+			},
+			"resources": [{
+					"type": "Microsoft.DataFactory/factories/linkedServices",
+					"name": "[concat(parameters('dataFactoryName'), '/ArmtemplateStorageLinkedService')]",
+					"apiVersion": "2018-06-01",
+            		"location": "[parameters('location')]",
+					"properties": {
+						"type": "AzureBlobStorage",
+						"typeProperties": {
+							"connectionString": "[concat('DefaultEndpointsProtocol=https;AccountName=',parameters('storageAccountName'),';AccountKey=',listKeys(variables('storageAccountId'), '2019-06-01').keys[0].value)]"
+						}
+					},
+					"dependsOn": [
```

Comment:
move all dependsOn properties above the properties property

Reply:
done. :)